include/envoy/config/subscription.h

```diff
@@ -146,12 +146,15 @@ class UntypedConfigUpdateCallbacks {
    * @param removed_resources names of resources that this fetch instructed to be removed.
    * @param system_version_info aggregate response data "version", for debugging.
    * @throw EnvoyException with reason if the config changes are rejected. Otherwise the changes
-   *        are accepted. Accepted changes have their version_info reflected in subsequent requests.
+   * @param use_namespace_matching if the resources should me matched on their namespaces, rather
+   * than unique names. This is used when a collection of resources (e.g. virtual hosts in VHDS) is
+   * being updated. are accepted. Accepted changes have their version_info reflected in subsequent
+   * requests.
    */
   virtual void onConfigUpdate(
       const Protobuf::RepeatedPtrField<envoy::service::discovery::v3::Resource>& added_resources,
       const Protobuf::RepeatedPtrField<std::string>& removed_resources,
-      const std::string& system_version_info) PURE;
+      const std::string& system_version_info, const bool use_namespace_matching) PURE;
```

Comment:
How come this needs to propagate all the way back to the subscriber? Shouldn't the subscriber already know if they are using namespace matching? I.e. VHDS knows that it is, EDS knows that it isn't.

Reply:
good call.